connection_tracker/bpf/conn_track.c

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#include "bpf_endian.h"
+#include "bpf_helpers.h"
+#include "net/ip.h"
+#include "..\conn_track\conn_tracker.h"
+
+inline __attribute__((always_inline)) uint64_t
+bpf_ntohll(uint64_t x)
+{
+    uint64_t upper = bpf_ntohl(x >> 32);
+    uint64_t lower = (uint64_t)bpf_ntohl(x & 0xFFFFFFFF) << 32;
+    return upper | lower;
+}
+
+#define bpf_htonll(x) bpf_ntohll(x)
+
+#ifndef ntohll
+#define ntohll bpf_ntohll
+#endif
+#ifndef htonll
+#define htonll bpf_htonll
+#endif
+
+typedef struct _connection_history
+{
+    connection_tuple_t tuple;
+    bool ipv4;
```

Comment:
nit (ok to punt to future PR): in my view it would be better to either name this `is_ipv4` or else store the address family value here instead of a boolean.